streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -215,14 +216,20 @@ synchronized void unlockGlobalState() throws IOException {
     synchronized void unlock(final TaskId taskId) throws IOException {
         final LockAndOwner lockAndOwner = locks.get(taskId);
         if (lockAndOwner != null && lockAndOwner.owningThread.equals(Thread.currentThread().getName())) {
-            locks.remove(taskId);
-            lockAndOwner.lock.release();
+            boolean isClosedChannelException = true;
+            try {
+                lockAndOwner.lock.release();
+            } catch (ClosedChannelException cce){
+                log.error("{} Closed Channel Exception Occured supposed to Ignore", logPrefix(), cce);
+                isClosedChannelException = false;
+            }
             log.debug("{} Released state dir lock for task {}", logPrefix(), taskId);
 
             final FileChannel fileChannel = channels.remove(taskId);
-            if (fileChannel != null) {
+            if (fileChannel != null && isClosedChannelException) {
```

Comment:
The javadoc suggests we don't have to handle this condition specially:
> If the channel has already been closed then this method returns immediately.

Reply:
@vvcephei - If I understood correctly, `isClosedChannelException` variable's initialization and check in `fileChannel != null` is not needed.

Reply:
Implemented Review Comment. Please review.